src/library-view-qt/preference.cc

```diff
@@ -0,0 +1,79 @@
+#include <glib.h>
+
+#include <libaudcore/audstrings.h>
+#include <libaudcore/runtime.h>
+#include <libaudqt/libaudqt.h>
+
+#include "library-view-qt.h"
+#include "tree-view.h"
+
+QPointer<PreferenceWidget> PreferenceWidget::instance;
+String prev_playlist_title;
+String prev_playing_title;
+
+PreferenceWidget::PreferenceWidget(QWidget* parent)
+	: QWidget(parent), box_layout(audqt::make_hbox(this)),
+	  file_entry(audqt::file_entry_new(this, _("Choose Folder"),
+					   QFileDialog::Directory,
+					   QFileDialog::AcceptOpen)) {
+	file_entry->setText(static_cast<const char*>(
+	filename_contract(uri_to_filename(get_pref_uri()))));
+	box_layout->addWidget(file_entry);
+	default_palette = file_entry->palette();
+	error_palette = default_palette;
+	error_palette.setColor(QPalette::Text, Qt::red);
```

Comment:
Errors are generally not colored in red like this in Audacious.  We prefer to use iconography, as it is more accessible.